api/v2/Routes/bins/index.js

```diff
@@ -1,9 +1,21 @@
 import { Router } from 'express';
+import { getBins } from "../../../lib/redisHelper.mjs";
+import NotFoundError from '../../../lib/HttpErrors/NotFoundError.js';
 
 const router = Router({ mergeParams: true });
 
 router.get('/', async (_, res) => {
-    return res.status(501).json({ message: 'Not implemented' });
+    try {
+        const binsData = await getBins();
+        
+        if (!binsData) {
+            throw new NotFoundError('Bins data not found');
+        }
+        return res.status(200).json(binsData['bins']);
```

Comment:
```suggestion
        return res.status(200).json(binsData.bins);
```
Since this is an attribute on an object, we should either use destructuring or dot notation.